src/basicfuns.jl

```diff
@@ -190,7 +190,13 @@ function logsumexp(X)
     isempty(X) && return log(sum(X))
     reduce(logaddexp, X)
 end
-function logsumexp(X::AbstractArray{T}) where {T<:Real}
+function logsumexp(X::AbstractArray{T}; dims=nothing) where {T<:Real}
+    dims isa Nothing && return _logsumexp(X)
```

Comment:
I think it's more contentional to make this something like
```suggestion
    dims === nothing && return _logsumexp(X)
```
though it doesn't really matter; they're entirely equivalent and should perform the same. Just figured I'd note it.

Reply:
`isnothing`?

Reply:
`===` is special cased in the compiler and is generally more efficient for checking `nothing` (and indeed, even `missing`).

Reply:
I think `isnothing` is also elided since it's dealt with at dispatch time, and it was the recommended way to check for nothingness? The only thing is that it requires Julia 1.1.

Reply:
> it was the recommended way to check for nothingness?

How did you come to that conclusion?

> I think `isnothing` is also elided

It is not, see https://github.com/JuliaLang/julia/issues/27681.

Reply:
> How did you come to that conclusion?

Well, it's a exported function named `isnothing` :trollface:
In reality I wasn't aware of that issue. Thanks for pointing it out.